dependency/actions/compile/entrypoint

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+function main() {
+  local version output_dir target httpd_dir
+  version=
+  output_dir=
+  target=
+
+  while [ "${#}" != 0 ]; do
+    case "${1}" in
+      --version)
+        version="${2}"
+        shift 2
+        ;;
+
+      --outputDir)
+        output_dir="${2}"
+        shift 2
+        ;;
+
+      --target)
+        target="${2}"
+        shift 2
+        ;;
+
+      "")
+        shift
+        ;;
+
+      *)
+        echo "unknown argument \"${1}\""
+        exit 1
+    esac
+  done
+
+  if [[ "${version}" == "" ]]; then
```

Comment:
You can simplify this conditional and remove the empty assignments on lines 8-10 if you just provide a default value when interpolating the variable. We use this pattern in a bunch of places, like `./scripts/package.sh` in a buildpack: https://github.com/paketo-buildpacks/github-config/blob/61d3ac234aaee0362463688ffb81403e501c24ef/implementation/scripts/package.sh#L47 

```suggestion
  if [[ -z "${version:-}" ]]; then
```

Reply:
Done.